### PR TITLE
Add restriction information for index merge to avoid misuse

### PR DESCRIPTION
**Pull request by ichn-hu**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add the restriction to avoid misuse.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: explain-index-merge.md [hunk 1]
```diff
@@ -43,3 +43,9 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
 > - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
 >
 > SQL Hint 的优先级高于系统变量。
+>
+> 此外，有两个限制会导致 `IndexMerge` 无法使用：
+> 
+> - 如果查询有除了全表扫以外的单索引扫描方式可以选择，优化器不会选择 index merge；
+> 
+> - 如果查询在显式事务里，且该条查询之前的语句已经涉及写入，优化器不会选择 index merge；
```
Comment (TomShawn) [inline explain-index-merge.md:51]:
```suggestion
> 此外，有以下两个限制会导致 `IndexMerge` 无法被使用：
> 
> - 如果查询有除了全表扫以外的单索引扫描方式可以选择，优化器不会选择 `IndexMerge` 。
> 
> - 如果查询在显式事务里，且该条查询之前的语句已经涉及写入操作，优化器不会选择 `IndexMerge` 。
```

**Comment by TomShawn** [comment]
@ichn-hu Could you please involve a technical reviewer? Thanks~

**Comment by ichn-hu** [comment]
@qw4990 PTAL

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.4 label because the v6.4 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.4 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.3 label because the v6.3 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.3 and will no longer receive new updates.

**Comment by hfxsd** [comment]
这些内容已经在最新的文档里有了。另外现在已经支持事务里面使用 index merge。

**Pull request closed without merging**